File: NodeJs Client/subscriber.js
/* ------------------------------------
Description : A publisher example to check will message
Broker: Mosquitto
*/

const mqtt = require('mqtt')
const TOPIC = 'cdac/diot/+/value'

// const BROKER_TOPIC = '$SYS/'
const WILL_TOPIC = 'dead/device'
//connection to broker
const client = mqtt.connect('mqtt://localhost:1883',{
    protocolVersion:5,
    keepalive:0,
    clientId:'cdacdiot_sensor',
    clean:false,
    username:'actuator',
    password:'qwerty123'
});

client.on('connect',()=>{
    console.log("i'm connected to the Broker")
    //Subscription to Broker on Topic
    client.subscribe(TOPIC,(err)=>{
        if(err){
            console.log("NOT connected properly");
            return;
        }
        console.log(`Subscriber to Topic: ${TOPIC}`)
    })
})

//To receive messages on subscribed TOPIC
client.on('message',(topic,payload)=>{
    console.log(`Message received at TOPIC: ${topic}`)
    console.log(payload.toString())
})

File: Python_clients/subscriber.py
import paho.mqtt.client as mqtt

subscriber = mqtt.Client()

TOPIC='cdac/diot'

# subscriber.username_pw_set("actuator","qwerty123")

def on_connection(Client,userdata,flags,rc):
    if rc== mqtt.MQTT_ERR_SUCCESS:
     print('connection with MQTT broker now')
     subscriber.subscribe(TOPIC)

    else:
       print("Error occured")

def on_receive(Client,userdata,msg):
   print(f'Topic : {msg.topic}, Message : {str(msg.payload)}')
  
  
subscriber.on_connect = on_connection
subscriber.on_message = on_receive


subscriber.connect("localhost",1883)

subscriber.loop_forever()
